pandas/io/sas/sasreader.py

```diff
@@ -136,8 +136,8 @@ def read_sas(
     if iterator or chunksize:
         return reader
 
-    data = reader.read()
-
-    if ioargs.should_close:
-        reader.close()
-    return data
+    try:
+        return reader.read()
```

Comment:
is this needed that you are catching error (and closing) at the lower level? maybe *only* catch here

Reply:
The close at the lower level (sas7bdat.py, sas_xport.py) is to match with the get_filepath_or_buffer call 10 lines earlier, and also with various calls to close scatter around those files.
It is possible (if both 'lower level' file are only ever used via sasreader.py) that all close calls are unnecessary lower down, but I don't think they do any harm so I would prefer to leave the code symmetrical for the moment (open/close) and potentially get rid of all the close calls in a separate PR.
What do you think?

Reply:
ok fair enough